src/wasm.h

```diff
@@ -176,6 +176,16 @@ struct Literal {
     }
   }
 
+  int32_t floatPayload() {
+    assert(type == WasmType::f32);
+    return ~0xffc00000u & bit_cast<uint32_t>(f32);
```

Comment:
Ignore my refsctor comment, I just saw that you modified the code below. That's what I get for reviewing kn a phone from the airport. 

I'd still assert the number is nan though.